Classes/Projection/ProjectionManager.php

```diff
@@ -0,0 +1,238 @@
+        $projectionsByName = $projectorClassNamesByIdentifier = [];
+        foreach ($reflectionService->getAllImplementationClassNamesForInterface(ProjectorInterface::class) as $projectorClassName) {
+            $package = $packageManager->getPackageByClassName($projectorClassName);
+            $projectionName = (new ClassReflection($projectorClassName))->getShortName();
+            if (substr($projectionName, -9) === 'Projector') {
+                $projectionName = substr($projectionName, 0, -9);
+            }
+            $projectionName = strtolower($projectionName);
+            $packageKey = strtolower($package->getPackageKey());
+            $projectionIdentifier = $packageKey . ':' . $projectionName;
+            if (isset($projectorClassNamesByIdentifier[$projectionIdentifier])) {
+                throw new \RuntimeException(sprintf('The projection identifier "%s" is ambiguous, please rename one of the classes "%s" or "%s"', $projectionIdentifier, $projectorClassNamesByIdentifier[$projectionIdentifier], $projectorClassName), 1476198478);
+            }
+            $projectorClassNamesByIdentifier[$projectionIdentifier] = $projectorClassName;
+            if (!isset($projectionsByName[$projectionName])) {
+                $projectionsByName[$projectionName] = [];
+            }
+            $projectionsByName[$projectionName][] = $packageKey;
+        }
+        ksort($projectorClassNamesByIdentifier);
+
+        $projections = [];
+        foreach ($projectorClassNamesByIdentifier as $fullProjectionIdentifier => $projectorClassName) {
+            list($packageKey, $projectionName) = explode(':', $fullProjectionIdentifier);
+            if (count($projectionsByName[$projectionName]) === 1) {
+                $shortIdentifier = $projectionName;
+            } else {
+                $shortIdentifier = $fullProjectionIdentifier;
+                $prefix = null;
+                foreach (array_reverse(explode('.', $packageKey)) as $packageKeyPart) {
+                    $prefix = $prefix === null ? $packageKeyPart : $packageKeyPart . '.' . $prefix;
+                    $matchingPackageKeys = array_filter($projectionsByName[$projectionName], function ($searchedPackageKey) use ($packageKey) {
+                        return $searchedPackageKey === $packageKey || substr($packageKey, -(strlen($searchedPackageKey) + 1)) === '.' . $searchedPackageKey;
+                    });
+                    if (count($matchingPackageKeys) === 1) {
+                        $shortIdentifier = $prefix . ':' . $projectionName;
+                        break;
+                    }
+                }
+            }
+            $projectionEventTypes = $eventListenerLocator->getEventTypesByListenerClassName($projectorClassName);
+            $projections[$fullProjectionIdentifier] = new Projection($fullProjectionIdentifier, $shortIdentifier, $projectorClassName, $projectionEventTypes);
+        }
```

Comment:
This is indeed quite a lot of complexity only to find the shortest unambiguous projection designator.. But I think it's a very important feature for the CLI (and that's the only place where it's used currently)